### PR TITLE
docs(new): migrate HTTPResponse docs to TSDoc

### PR DESCRIPTION
**Pull request by hanselfmu**

### CODE REVIEW

File: src/common/HTTPResponse.ts [hunk 1]
```diff
@@ -1,29 +1,37 @@
...
  *
  * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
```
Comment (jackfranklin) [inline src/common/HTTPResponse.ts]: Did you rewrap this bit? Could we revert this diff just so the license is consistent in all the files? :)

File: src/common/HTTPResponse.ts [hunk 2]
```diff
@@ -67,38 +78,70 @@ export class HTTPResponse {
...
+   */
   url(): string {
     return this._url;
   }
 
+  /**
+   * @returns True if the response was successful (status in the range 200-299).
+   */
   ok(): boolean {
+    // TODO: document === 0 case?
```
Comment (jackfranklin) [inline src/common/HTTPResponse.ts:108]: @mathiasbynens do you have any context to this case? I'm not sure why we'd have this.

File: src/common/HTTPResponse.ts [hunk 3]
```diff
@@ -1,29 +1,37 @@
...
+/**
+ * @public
+ */
+export interface RemoteAddress {
   ip: string;
   port: number;
 }
 
+/**
+ * The HTTPResponse class represents responses which are received by page.
```
Comment (jackfranklin) [inline src/common/HTTPResponse.ts]: Maybe we could link to the page class here? `responses which are received by instances of the {@link Page} class`

**Review by jackfranklin — COMMENTED** [review]
Minor comments, looks great on the whole, thank you!

File: src/common/HTTPResponse.ts (hunk 2 above)
Comment (mathiasbynens) [reply src/common/HTTPResponse.ts:108]: I don't know, but I would _guess_ this is used for errors before we can even get an HTTP response + status code, e.g. certificate issues. Maybe @sigurdschneider knows?

File: src/common/HTTPResponse.ts (hunk 1 above)
Comment (hanselfmu) [reply src/common/HTTPResponse.ts]: Yes I did. Let me change this part back. Enough for the laziness of cmd+A and alt+Q I guess :P

File: src/common/HTTPResponse.ts (hunk 3 above)
Comment (hanselfmu) [reply src/common/HTTPResponse.ts]: Yes and I like the minor tweak of the grammar.

**Review by jackfranklin — APPROVED** [review]
This looks good to me ! If we're not sure on the status code 0 case, happy to merge as is.

**Comment by jackfranklin** [comment]
I'm going to merge this, and we can make another PR to address the status = 0 case.

**Pull request merged**